core/trino-spi/src/main/java/io/trino/spi/connector/RecordCursor.java

```diff
@@ -55,4 +72,11 @@ default long getSystemMemoryUsage()
 
     @Override
     void close();
+
+    enum AdvanceStatus
```

Comment:
Adding yielding without possibility of blocking leads to interface that can active loop. This should be modeled after `ConnectorPageSource`